lms/djangoapps/django_comment_client/utils.py

```diff
@@ -69,31 +79,67 @@ def get_accessible_discussion_modules(course, user, include_all=False):  # pylin
     """
     all_modules = modulestore().get_items(course.id, qualifiers={'category': 'discussion'})
 
-    def has_required_keys(module):
-        for key in ('discussion_id', 'discussion_category', 'discussion_target'):
-            if getattr(module, key, None) is None:
-                log.warning("Required key '%s' not in discussion %s, leaving out of category map" % (key, module.location))
-                return False
-        return True
-
     return [
         module for module in all_modules
         if has_required_keys(module) and (include_all or has_access(user, 'load', module, course.id))
     ]
 
 
+def get_discussion_id_map_entry(module):
+    """
+    Returns a tuple of (discussion_id, metadata) suitable for inclusion in the results of get_discussion_id_map().
+    """
+    discussion_id = module.discussion_id
+    title = module.discussion_target
+    last_category = module.discussion_category.split("/")[-1].strip()
+    return (discussion_id, {"location": module.location, "title": last_category + " / " + title})
```

Comment:
nitpick: you could inline all the dictionary values instead of creating variables for them, since they are only read back out once.